src/PJ_merc.c

```diff
@@ -76,3 +119,22 @@ PJ *PROJECTION(merc) {
     return P;
 }
 
+PJ *PROJECTION(webmerc) {
+
+    if( pj_param(P->ctx, P->params, "tk_0").i ) {
```

Comment:
This behaviour is different to the rest of PROJ. Usually you are allowed to set a parameter even if it is not used. If it is not used by the projection it will just get ignored. `proj` will tell you if that is the case:

```
$ proj -v +proj=merc +lat_1=23
#Mercator
#       Cyl, Sph&Ell
#       lat_ts=
# +proj=merc +ellps=WGS84
#--- following specified but NOT used
# +lat_1=23
```

I think it would be better to keep this "tradition". Just make sure to override them in the setup with `P->k0=1.0; P->lam0=0.0`.

Reply:
I kept the ones for k_0 and lat_0 since those are done in generic parts. I remove the one for lat_ts (which was wrongly spelled in my original commit)

Reply:
I understand why you would want to do that but I still don't think it is the correct way to handle it. I think it will cause more confusion than is necessary. This would be the only place in the code base where using one of the default projection parameters would cause an error. There are plenty [examples](https://github.com/OSGeo/proj.4/blob/master/src/PJ_stere.c#L295-L314) of [overriding](https://github.com/OSGeo/proj.4/blob/master/src/PJ_nzmg.c#L108-L113) default [parameters](https://github.com/OSGeo/proj.4/blob/master/src/PJ_krovak.c#L185-L186) in the setup without raising errors.

I think it would be awesome if we had a system that caused errors when using parameters that can't be used with a given operation but that is a problem for another day (and major version release :-) ).

Reply:
OK, checks reverted and parameters overriden